geomesa-blobstore/geomesa-blobstore-core/src/main/scala/org/locationtech/geomesa/blob/core/AccumuloBlobStore.scala

```diff
@@ -0,0 +1,112 @@
+package org.locationtech.geomesa.blob.core
+
+import java.io.File
+
+import com.google.common.io.{ByteStreams, Files}
+import org.apache.accumulo.core.client.admin.TimeType
+import org.apache.accumulo.core.client.{Scanner, TableExistsException}
+import org.apache.accumulo.core.data.{Key, Mutation, Range, Value}
+import org.apache.accumulo.core.security.Authorizations
+import org.apache.hadoop.io.Text
+import org.geotools.data.Query
+import org.geotools.data.collection.ListFeatureCollection
+import org.geotools.data.simple.SimpleFeatureStore
+import org.locationtech.geomesa.accumulo.data.{AccumuloDataStore, _}
+import org.locationtech.geomesa.accumulo.util.{GeoMesaBatchWriterConfig, SelfClosingIterator}
+import org.locationtech.geomesa.blob.core.AccumuloBlobStore._
+import org.locationtech.geomesa.blob.core.handlers.BlobStoreFileHandler
+import org.locationtech.geomesa.utils.geotools.Conversions._
+import org.locationtech.geomesa.utils.geotools.SimpleFeatureTypes
+import org.opengis.feature.simple.SimpleFeatureType
+import org.opengis.filter.Filter
+
+import scala.collection.JavaConversions._
+
+class AccumuloBlobStore(ds: AccumuloDataStore) {
+
+  private val connector = ds.connector
+  private val tableOps = connector.tableOperations()
+
+  val blobTableName = s"${ds.catalogTable}_blob"
+
+  ensureTableExists(blobTableName)
+  ds.createSchema(sft)
+  val bw = connector.createBatchWriter(blobTableName, GeoMesaBatchWriterConfig())
+  val scanner: Scanner = connector.createScanner(blobTableName, new Authorizations())
+
+  val fs = ds.getFeatureSource(blobFeatureTypeName).asInstanceOf[SimpleFeatureStore]
+
+  def put(file: File, params: Map[String, String]): Option[String] = {
+    BlobStoreFileHandler.buildSF(file, params).map {
+      sf =>
+        val id = sf.getAttribute(idFieldName).asInstanceOf[String]
+
+        fs.addFeatures(new ListFeatureCollection(sft, List(sf)))
+        putInternal(file, id)
+        id
+    }
+  }
+
+  def getIds(filter: Filter): Iterator[String] = {
+    getIds(new Query(blobFeatureTypeName, filter))
+  }
+
+  def getIds(query: Query): Iterator[String] = {
+    fs.getFeatures(query).features.map(_.getAttribute(idFieldName).asInstanceOf[String])
+  }
+
+  def get(id: String): (Array[Byte], String) = {
+    scanner.setRange(new Range(new Text(id)))
+
+    val iter = SelfClosingIterator(scanner)
+    if (iter.hasNext) {
+      val ret = buildReturn(iter.next)
+      iter.close()
+      ret
+    } else {
+      (Array.empty[Byte], "")
+    }
+  }
+
+  def buildReturn(entry: java.util.Map.Entry[Key, Value]): (Array[Byte], String) = {
+    val key = entry.getKey
+    val value = entry.getValue
+
+    val filename = key.getColumnQualifier.toString
+
+    (value.get, filename)
+  }
+
+  def putInternal(file: File, id: String) {
+    val localName = file.getName
+    val bytes =  ByteStreams.toByteArray(Files.newInputStreamSupplier(file))
```

Comment:
if the file is big, this could blow up. I remember seeing a blob store implementation in https://github.com/calrissian/accumulo-recipes that gets around the issue.